HARK/ConsumptionSaving/ConsIndShockModel.py

```diff
@@ -2659,77 +2660,72 @@ def construct_lognormal_income_process_unemployment(self):
             a discrete approximation to the transitory income shocks.
         """
         # Unpack the parameters from the input
+        T_cycle = self.T_cycle
         PermShkStd = self.PermShkStd
         PermShkCount = self.PermShkCount
         TranShkStd = self.TranShkStd
         TranShkCount = self.TranShkCount
-        T_cycle = self.T_cycle
         T_retire = self.T_retire
         UnempPrb = self.UnempPrb
         IncUnemp = self.IncUnemp
         UnempPrbRet = self.UnempPrbRet
         IncUnempRet = self.IncUnempRet
 
-        IncShkDstn = []  # Discrete approximations to income process in each period
-        PermShkDstn = []  # Discrete approximations to permanent income shocks
-        TranShkDstn = []  # Discrete approximations to transitory income shocks
-
-        # Fill out a simple discrete RV for retirement, with value 1.0 (mean of shocks)
-        # in normal times; value 0.0 in "unemployment" times with small prob.
         if T_retire > 0:
-            if UnempPrbRet > 0:
-                PermShkValsRet = np.array(
-                    [1.0, 1.0]
-                )  # Permanent income is deterministic in retirement (2 states for temp income shocks)
-                TranShkValsRet = np.array(
-                    [
-                        IncUnempRet,
-                        (1.0 - UnempPrbRet * IncUnempRet) / (1.0 - UnempPrbRet),
-                    ]
-                )
-                ShkPrbsRet = np.array([UnempPrbRet, 1.0 - UnempPrbRet])
-            else:
-                PermShkValsRet = np.array([1.0])
-                TranShkValsRet = np.array([1.0])
-                ShkPrbsRet = np.array([1.0])
-            IncShkDstnRet = DiscreteDistribution(
-                ShkPrbsRet,
-                [PermShkValsRet, TranShkValsRet],
-                seed=self.RNG.randint(0, 2 ** 31 - 1),
-            )
+            normal_length = T_retire
+            retire_length = T_cycle - T_retire
+        else:
+            normal_length = T_cycle
+            retire_length = 0
+
+        UnempPrb_list = [UnempPrb]*normal_length + [UnempPrbRet]*retire_length
+        IncUnemp_list = [IncUnemp]*normal_length + [IncUnempRet]*retire_length
```

Comment:
Do you see a clear path from this to a solution to #1015 ?

Reply:
Yes, absolutely. That was my motivation for doing this, and it would amount to simply allowing users to pass these lists (as they do for the standard deviations) instead of floats. I just wanted to get what is currently there updated before introducing that feature.